dashboard/src/router/UseGlobalNavGuards.js

```diff
@@ -57,25 +57,33 @@ export const useGlobalNavGuards = () => {
   const isPki = () => appConfig.isPkiAuthenticated
   const getLandingPage = () => authState.userInfo?.landingPage === 'admin' ? 'AdminHomePage' : 'MyProgressPage'
 
+  const registrationId = () => appConfig.saml2RegistrationId;
+  const isSaml2 = () => appConfig.isSAML2Authenticated;
+
   const beforeEachNavGuard = (to, from, next) => {
     const { skillsClientDisplayPath } = to.query
     const requestAccountPath = '/request-root-account'
     const skillsLoginPath = '/skills-login'
+    const saml2LoginPath = '/sso-login'
+   
+    console.log('From :' + from.path + ' To :'+to.path)
+    console.log('Is Logged In :: '+isLoggedIn())
+
     if (
-      !isPki() &&
+      !isPki() && !isSaml2() &&
       !isLoggedIn() &&
       to.path?.toLowerCase() !== requestAccountPath &&
       appConfig.needToBootstrap
     ) {
       next({ path: requestAccountPath })
     } else if (
-      !isPki() &&
-      to.path?.toLowerCase() === requestAccountPath &&
+      !isPki() && !isSaml2() &&
```

Comment:
- `.toLowerCase()` was removed; perhaps it would be better to add `toLowerCase()` to `requestAccountPath` to make the comparison case insensitive? 
- && `!isSaml2()` is not needed either here; we plan to remove the `!isPki()` check in the future

Reply:
Refactored as per your comments